arbitrary-granularity/src/main/java/org/apache/druid/java/util/common/granularity/ArbitraryGranularity.java

```diff
@@ -108,56 +111,69 @@ public ArbitraryGranularity(List<Interval> inputIntervals)
   @Override
   public boolean isAligned(Interval interval)
   {
-    throw new UnsupportedOperationException(
-      "This method should not be invoked for this granularity type"
-    );
+    throw new UnsupportedOperationException(METHOD_NOT_SUPPORTED_MSG);
   }
 
   // Used only for Segments. Not for Queries
   @Override
   public DateTimeFormatter getFormatter(Formatter type)
   {
-    throw new UnsupportedOperationException(
-      "This method should not be invoked for this granularity type"
-    );
+    throw new UnsupportedOperationException(METHOD_NOT_SUPPORTED_MSG);
   }
 
   // Used only for Segments. Not for Queries
   @Override
   public DateTime toDate(String filePath, Formatter formatter)
   {
-    throw new UnsupportedOperationException(
-      "This method should not be invoked for this granularity type"
-    );
+    throw new UnsupportedOperationException(METHOD_NOT_SUPPORTED_MSG);
   }
 
   @Override
   public long increment(long time)
   {
-    // Not sure how else to accomplish this with arbitrary intervals. toDateTime(time) defeats the purpose of #10904
-    return increment(toDateTime(time)).getMillis();
+    // Test if the input cannot be bucketed
+    if (time >= intervals.last().getEndMillis()) {
+      return MAX_DATETIME.getMillis();
+    }
+
+    // First interval with start time <= timestamp
+    final Interval interval = intervals.floor(new Interval(time, MAX_DATETIME.getMillis()));
```

Comment:
This Interval constructor is forbidden in upstream code base. Probably should take care of it here. See [this](https://github.com/apache/druid/blob/master/codestyle/joda-time-forbidden-apis.txt#L45) (same applies on line 170). `timezone` is available as an instance variable and that constructor is not forbidden.. just a thought, not sure if it is right answer here.

Reply:
Good catch Lucas - I fixed it.